services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/LinkServiceImpl.java

```diff
@@ -156,4 +164,21 @@ private BatchResults getLinkPropsResult(List<LinkPropsResponse> responses, List<
 
         return new BatchResults(errors.size(), successCount, errors);
     }
+
+    @Override
+    public CompletableFuture<List<FlowPayload>> getFlowsForLink(SwitchId srcSwitch, Integer srcPort,
+                                                                SwitchId dstSwitch, Integer dstPort) {
+        final String correlationId = RequestCorrelationId.getId();
+        logger.debug("Get all flows for a particular link request processing");
+        GetFlowsForLinkRequest data = new GetFlowsForLinkRequest(new NetworkEndpointMask(srcSwitch, srcPort),
```

Comment:
Why is NetworkEndpointMask used instead of NetworkEndpoint? As far as I see both switchId and port is required.

Reply:
Fixed.